homeassistant/components/modbus/base_platform.py

```diff
@@ -76,7 +77,9 @@ class BasePlatform(Entity):
     def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
         """Initialize the Modbus binary sensor."""
         self._hub = hub
-        self._slave = entry.get(CONF_SLAVE, 0)
+        self._slave = entry.get(CONF_SLAVE, None)
+        if not self._slave:
+            self._slave = entry.get(CONF_DEVICE_ADDRESS, 0)
```

Comment:
```suggestion
        self._slave = entry.get(CONF_SLAVE, None) or entry.get(CONF_DEVICE_ADDRESS, 0)
```
You can do the same here

Reply:
missed that done, but in a slightly different manner, that at least to me adds less confusion.